addons/l10n_vn/migrations/17.0.2.0.2/post-migration.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+from odoo import api, SUPERUSER_ID
+
+FIXED_ACCOUNTS_TYPE = {
+    '242': 'asset_prepayments',
+    '6274': 'expense_depreciation',
+    '6414': 'expense_depreciation',
+    '6424': 'expense_depreciation',
+}
+
+
+def _fix_accounts_type(env):
+    for company in env['res.company'].with_context(active_test=False).search([('chart_template', '=', 'vn')]):
+        for code, correct_type in FIXED_ACCOUNTS_TYPE.items():
+            account = env['account.account'].search([
+                ('code', '=', code), ('company_id', '=', company.id), ('account_type', '!=', correct_type)
+            ])
+            if account:
+                account.write({'account_type': correct_type})
```

Comment:
I think it would be a better idea to retrieve the accounts using `env.ref(external_id)` instead.
It would be more precise.

Also, not important in this context but worth mentioning: writing in batch on the accounts having the same account_type would be better for the sake of optimization.